supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiBackendLifecycleActorFactory.scala

```diff
@@ -82,4 +90,48 @@ abstract class PipelinesApiBackendLifecycleActorFactory(override val name: Strin
 object PipelinesApiBackendLifecycleActorFactory {
   val preemptionCountKey = "PreemptionCount"
   val unexpectedRetryCountKey = "UnexpectedRetryCount"
+
+  private val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  private [common] def robustBuildAttributes(buildAttributes: () => PipelinesApiConfigurationAttributes,
+                                             maxAttempts: Int = 3,
+                                             initialIntervalMillis: Int = 5000,
+                                             maxIntervalMillis: Int = 10000,
+                                             multiplier: Double = 1.5,
+                                             randomizationFactor: Double = 0.5): PipelinesApiConfigurationAttributes = {
+    val backoff = new ExponentialBackOff.Builder()
+      .setInitialIntervalMillis(initialIntervalMillis)
+      .setMaxIntervalMillis(maxIntervalMillis)
+      .setMultiplier(multiplier)
+      .setRandomizationFactor(randomizationFactor)
+      .build()
+
+    // Is this an `Exception` (as opposed to an `Error`) with a message indicating the operation should be retried?
+    def isRetryableException(t: Throwable): Boolean = {
+      t match {
+        case e: Exception =>
+          Option(e.getMessage) match {
+            case Some(message) => message.contains("We encountered an internal error. Please try again.")
+            case None => false
+          }
+        case _ => false
+      }
+    }
+
+    // `attempt` is 1-based
+    @tailrec
+    def build(attempt: Int): PipelinesApiConfigurationAttributes = {
```

Comment:
Make `private`?

Reply:
this is a local function, more private than `private`